dkist/wcs/models.py

```diff
@@ -753,27 +763,56 @@ def __init__(self, array_shape, order='C', **kwargs):
         self.order = order
 
     def evaluate(self, *inputs):
-        ravel_shape = self.array_shape[1]
-        if self.order == 'F':
-            inputs = inputs[::-1]
-            ravel_shape = self.array_shape[0]
-        return (np.round(inputs[0]) * ravel_shape) + inputs[1]
+        # If we have units, then all values must have units, hence test for existence only on the first one.
+        # Also, assume all units are the same.
+        if hasattr(inputs[0], "unit"):
+            input_units = inputs[0].unit
+            input_values = [item[0].value for item in inputs]
+        else:
+            input_units = None
+            input_values = [item[0] for item in inputs]
+        # round the index values, but clip them if they exceed the array bounds
+        # the bounds are one lass than the shape dimension value
+        array_bounds = tuple(np.array(self.array_shape) - 1)
+        # array_bounds = [item - 1 for item in self.array_shape]
+        rounded_inputs = np.clip(np.rint(input_values).astype(int), None, array_bounds)
+        result = np.ravel_multi_index(rounded_inputs, self.array_shape, order=self.order).astype(float)
```

Comment:
Why the `astype(float)` here?

Reply:
Because the input coordinates are floating point and we round all but the inner-most coordinate (inner in the sense of memory access) so that we can pass the interpolation on to Tabular1D.